src/Areas/AzureArc/Services/ArcServices.cs

```diff
@@ -0,0 +1,298 @@
+
+        public async Task<DeploymentResult> OnboardClusterToArcAsync(string clusterName, string resourceGroupName, string location, string subscriptionId, string tenantId, string kubeConfigPath, string userProvidedPath)
+        {
+            string tempScriptPath = Path.Combine(userProvidedPath, "OnboardClusterToArc.ps1");
+            File.WriteAllText(tempScriptPath, LoadResourceFiles(OnboardClusterToArc));
+
+            var arguments = $"-ExecutionPolicy Bypass -File \"{tempScriptPath}\" -Force";
+
+            if (!string.IsNullOrEmpty(clusterName))
+                arguments += $" -ClusterName {clusterName}";
+            if (!string.IsNullOrEmpty(resourceGroupName))
+                arguments += $" -ResourceGroupName {resourceGroupName}";
+            if (!string.IsNullOrEmpty(subscriptionId))
+                arguments += $" -SubscriptionId {subscriptionId}";
+            if (!string.IsNullOrEmpty(tenantId))
+                arguments += $" -TenantId {tenantId}";
+            if (!string.IsNullOrEmpty(location))
+                arguments += $" -Location {location}";
+            if (!string.IsNullOrEmpty(kubeConfigPath))
+                arguments += $" -KubeConfigPath \"{kubeConfigPath}\"";
+
+            var processStartInfo = new ProcessStartInfo
+            {
+                FileName = "powershell.exe",
+                Arguments = arguments,
+                UseShellExecute = true,
+                Verb = "runas"
+            };
+
+            Process process = StartProcess(tempScriptPath, processStartInfo);
+            await Task.Run(() => process.WaitForExit());
+
+            return new DeploymentResult
+            {
+                Success = process.ExitCode == 0,
+                Steps = process.ExitCode == 0 ? "Arc Onboarding completed successfully." : "Arc Onboarding failed."
+            };
+
+        }
+        public async Task<DeploymentResult> DisconnectFromAzureArcAsync(string resourceGroupName, string clusterName, string userProvidedPath)
+        {
+            string tempScriptPath = Path.Combine(userProvidedPath, "DisconnectFromAzureArc.ps1");
+            File.WriteAllText(tempScriptPath, LoadResourceFiles(DisconnectFromAzureArc));
+
+            var arguments = $"-ExecutionPolicy Bypass -File \"{tempScriptPath}\" -Force";
+
+            if (!string.IsNullOrEmpty(resourceGroupName))
+                arguments += $" -ResourceGroupName {resourceGroupName}";
+            if (!string.IsNullOrEmpty(clusterName))
+                arguments += $" -ClusterName {clusterName}";
+
+            var processStartInfo = new ProcessStartInfo
+            {
+                FileName = "powershell.exe",
+                Arguments = arguments,
+                UseShellExecute = true,
+                Verb = "runas"
+            };
+
+            Process process = StartProcess(tempScriptPath, processStartInfo);
+            await Task.Run(() => process.WaitForExit());
```

Comment:
since we need to prepare and run script in all service methods, does it make sense to use the pattern

```
1. arguments = preparePSScript(userProvidedPath, scriptName);
2. add any additional args to arguments
3. int status = runPSScript(arguments) // this method will create process, execute and return exit code.
```